### DISCUSSION TITLE
GitHub CLI no longer has access to Codespaces commands within the Codespace

### DISCUSSION THREAD

**Discussion by sayhiben** [community/community#16049] [Codespaces] [answered]
My team previously used the GitHub CLI to list open ports from within the Codespace, however this stopped working within the past few weeks:

```shell
@vscode /workspaces/web - (master) $ gh cs ports
you have no codespaces
```

This command works locally, authenticated as the same GitHub user. It's unclear to me if the provisioned `GITHUB_TOKEN` has changed such that it no longer has the Codespaces scope, but my attempts to replace the token with `unset GITHUB_TOKEN && gh auth login` seem ineffective, so I'm unsure what the cause is. Any ideas?

**Comment by sayhiben** [comment]
Follow-up: I believe what's broken is _listing_ Codespaces. I can run the following command successfully within the Codespace:

```shell
gh cs ports -c $CODESPACE_NAME
```

**Comment by dmilburn** [comment, marked as answer]
Thanks so much for reporting and for doing some further investigation. I'm sorry for the inconvenience! I've added an internal issue to track this bug fix

**Comment by dlemstra** [comment]
It would also be useful if we could use the cli to rename the display name of the current code space. But that also does not seem to work:

```
@dlemstra ➜ /workspaces/ImageMagick (main ✗) $ gh cs edit
must provide `--display-name` or `--machine`

Usage:  gh codespace edit [flags]

Flags:
  -c, --codespace string      Name of the codespace
  -d, --display-name string   Set the display name
  -m, --machine string        Set hardware specifications for the VM
  
@dlemstra ➜ /workspaces/ImageMagick (main ✗) $ gh cs edit -d test
you have no codespaces
```

**Reply by sayhiben** [reply]
[It's in the environment](https://docs.github.com/en/codespaces/developing-in-codespaces/default-environment-variables-for-your-codespace#list-of-default-environment-variables), see `echo $CODESPACE_NAME`

**Reply by dlemstra** [reply]
That's the current value but it would be nice to be able to change name of the codespace with the cli.
